provisioner/shell/provisioner.go

```diff
@@ -238,7 +238,7 @@ func (p *Provisioner) Provision(ctx context.Context, ui packer.Ui, comm packer.C
 
 		// upload the var file
 		var cmd *packer.RemoteCmd
-		err = retry.Config{StartTimeout: p.config.startRetryTimeout}.Run(ctx, func(ctx context.Context) error {
+		return retry.Config{StartTimeout: p.config.startRetryTimeout}.Run(ctx, func(ctx context.Context) error {
```

Comment:
This will also return in case there's an error; I don't think this is what we want here.

Reply:
You're right. Fixed and squashed.